### PR TITLE
GTK4 enhancements

### PR DESCRIPTION
**Pull request by leapofazzam123**

### CODE REVIEW

**Comment by leapofazzam123** [comment]
When merge

**Comment by suhan-paradkar** [comment]
Also... Can you enable wayland backend too??

**Comment by leapofazzam123** [comment]
> Also... Can you enable wayland backend too??

Sure, do i need to add wayland dependencies to dependency list

**Comment by suhan-paradkar** [comment]
Yes.. the depenencies are already present in the repo.. hopefully all you need to is to add them to `TERMUX_PKG_DEPENDS`

**Comment by leapofazzam123** [comment]
Ok

**Comment by leapofazzam123** [comment]
Can i also enable vulkan support?

**Comment by suhan-paradkar** [comment]
Sure.. go ahead... I had disabled it when vulkan wasn't in the repo

**Comment by leapofazzam123** [comment]
Where do i need to put vulkan-headers? In build dependency or runtime dependency?

**Comment by suhan-paradkar** [comment]
Since those are just headers... Better place them as a build dependency

**Comment by leapofazzam123** [comment]
Ok

**Comment by suhan-paradkar** [comment]
Well... It uses the wrong linker...

**Comment by leapofazzam123** [comment]
How do i fix it

**Comment by leapofazzam123** [comment]
@suhan-paradkar

**Comment by suhan-paradkar** [comment]
Well it seems like it rebuilds wayland-protocols again...

> Dependency wayland-protocols found: NO found 1.17 but need: '>= 1.21'

You need to either bump wayland-protocols to v1. 21 or you need to patch meson.build file to use v1.17

**Comment by leapofazzam123** [comment]
Ok

**Comment by leapofazzam123** [comment]
I bumped wayland-protocols right in this pr

**Comment by leapofazzam123** [comment]
It's probably better to do it on serperate pr

File: packages/gtk4/build.sh [hunk 1]
```diff
@@ -3,19 +3,21 @@ TERMUX_PKG_DESCRIPTION="GObject-based multi-platform GUI toolkit"
...
+-Dintrospection=disabled
+-Dwayland-backend=true
+-Ddemos=true
 -Dbuild-examples=false
 -Dbuild-tests=false
 -Dvulkan=disabled
 -Dprint-cups=disabled
 -Dprint-cloudprint=disabled
 "
+LDFLAGS=-landroid-shmem
```
Comment (thunder-coding) [inline packages/gtk4/build.sh:23]: `LDFLAGS+=` instead of `LDFLAGS=`

**Review by thunder-coding — CHANGES_REQUESTED** [review]

File: packages/gtk4/build.sh (hunk 1 above)
Comment (leapofazzam123) [reply packages/gtk4/build.sh]: Does it really resolves the undefined reference problem?

Comment (thunder-coding) [reply packages/gtk4/build.sh]: Seems like the build files aren't respecting `LDFLAGS` 🤔

**Comment by suhan-paradkar** [comment]
It must be `LDFLAGS+="-landroid-shmem" `
Use double quotes...

**Comment by WMCB-Tech** [comment]
for this one:
https://github.com/termux/x11-packages/pull/433/checks?check_run_id=3620104691#step:4:5202
            
It looks like you have to add [`shm_open` and `shm_unlink functions](https://github.com/search?q=org%3Atermux+shm_open&type=code) in `../src/gdk/wayland/gdkdisplay-wayland.c`

**Pull request closed without merging**